HomeWork/Home2.2/main.java

```diff
@@ -0,0 +1,58 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Scanner;
+
+// the most concerns are the same as for C++ - we can intrduce piixel class and Image class and rework code to have  better isolatiion and encapsulation
+public class main{
+
+    // Function to process the input file and generate the output file
+    public static void processFile(String inputFileName, String outputFileName, int[] favoriteColor, int[] unfavoriteColor) {
```

Comment:
Encapsulation is needed instead of long function with multiple responsibilities - Pixel, Image classes will allow to organise code in more readable and clean way